File: Daily Challenge/Java/String - Substring Count 16.4.2026.java
import java.util.*;

public class Hello {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String s = sc.next();
        String e = sc.next();

        int cnt = 0;

        for(int i = 0; i < s.length(); i++) {
            for(int j = i; j < s.length(); j++) {
                String sub = s.substring(i, j + 1);
                if(sub.equals(e)) {
                    cnt++;
                }
            }
        }

        System.out.println(cnt);
    }
}


//2nd
import java.util.*;

public class Hello {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String a = sc.next();
        String b = sc.next();

        int count = 0;
        int index = 0;

        while((index = a.indexOf(b, index)) != -1) {
            count++;
            index++;  // move forward (for overlapping)
        }

        System.out.println(count);
    }
}

File: Daily Challenge/Python/String - Substring Count 16.4.2026.py
s = input().strip()
e = input().strip()

cnt = 0

for i in range(len(s)):
    for j in range(i, len(s)):
        if s[i:j+1] == e:
            cnt += 1

print(cnt)


//2nd
a = input().strip()
b = input().strip()

count = 0
index = 0

while True:
    index = a.find(b, index)
    if index == -1:
        break
    count += 1
    index += 1   # move forward for overlapping

print(count)